core_backend/app/question_answer/speech_components/utils.py

```diff
@@ -0,0 +1,68 @@
+import os
+
+from pydub import AudioSegment
+
+from ...llm_call.llm_prompts import IdentifiedLanguage
+from ...utils import setup_logger
+
+logger = setup_logger("Voice utils")
+
+
+def get_gtts_lang_code_and_model(
+    identified_language: IdentifiedLanguage,
+) -> tuple[str, str]:
+    """
+    Maps IdentifiedLanguage values to Google Cloud Text-to-Speech language codes
+    and voice model names.
+    """
+    # Add language codes and voice models according to
+    # https://cloud.google.com/text-to-speech/docs/voices
+    mapping = {
+        IdentifiedLanguage.ENGLISH: ("en-US", "Neural2-D"),
+        # IdentifiedLanguage.SWAHILI: ("sw-TZ", "Neural2-D"), # no support for swahili
+        IdentifiedLanguage.HINDI: ("hi-IN", "Neural2-D"),
+        # Add more languages and models as needed
+    }
+
+    result = mapping.get(identified_language)
+    if result is None:
+        raise ValueError(f"Unsupported language: {identified_language}")
+
+    return result
+
+
+def convert_audio_to_wav(input_filename: str) -> str:
+    """
+    Converts an MP3 or M4A file to a WAV file and ensures the WAV file has
+    the required specifications.
+    """
+    file_extension = input_filename.lower().split(".")[-1]
+
+    if file_extension in ["mp3", "m4a"]:
+        audio = AudioSegment.from_file(input_filename, format=file_extension)
+
+        wav_filename = os.path.splitext(input_filename)[0] + ".wav"
+        audio.export(wav_filename, format="wav")
+        logger.info(f"Conversion complete. Output file: {wav_filename}")
+    else:
+        wav_filename = input_filename
+        logger.info(f"{wav_filename} is already in WAV format.")
```

Comment:
You should probably handle the case where it is not wav. Or are we assuming that if it's neither in mp3 or m4a that means it is in wav by default?

Reply:
yeah i was thinking that these 3 are the major types anyways right? are there any others we should consider?

Reply:
With the current code, an mp4 file would be considered as wav. We should handle the case where it is not part of any of the 3 format and  probably return an error.

Reply:
makes sense, i will do that